readthedocs/doc_builder/director.py

```diff
@@ -192,22 +192,13 @@ def build(self):
         3. build PDF
         4. build ePub
         """
-        config = self.data.config
-
         self.run_build_job("pre_build")
 
         # Build all formats
-        build_overridden = config.build.jobs.build is not None
-        if build_overridden:
-            self.run_build_job("build.html")
-            self.run_build_job("build.htmlzip")
-            self.run_build_job("build.pdf")
-            self.run_build_job("build.epub")
-        else:
-            self.build_html()
-            self.build_htmlzip()
-            self.build_pdf()
-            self.build_epub()
+        self.build_html()
+        self.build_htmlzip()
+        self.build_pdf()
+        self.build_epub()
```

Comment:
I find the implementation proposal I made in this comment https://github.com/readthedocs/readthedocs.org/pull/11710#discussion_r1820546420 a lot easier to follow than the current implementation.

With the current implementation, **we are running the commands** in a conditional statement by doing:

```python
if self.run_build_job("build.html"):
   return True
```

which is an anti-pattern to me -- a conditional should not run all the commands.

Reply:
Well, dynamically calling functions make code harder to navigate, the current implementation encapsulates all operations in a single method. Anyway, looks like more of a personal preference.